### DISCUSSION TITLE
Storage API: PUT 'the request is not multipart'

### DISCUSSION THREAD

**Discussion by leo-petrucci** [supabase/supabase#2051] [Questions] [answered]
I'm so sorry, I know I just opened a discussion but as soon as I fixed that I ended up bumping into another problem with the same endpoint but with the `PUT` method.

```
        fetch(`${SUPABASE_URL}/storage/v1/object/avatars/${name}`, {
          method: 'PUT',
          mode: 'cors',
          cache: 'no-cache',
          headers: {
            'Content-Type': file.type,
            'Content-Length': file.size.toString(),
            Authorization: `Bearer ${SUPABASE_SERVICE}`,
          },
          body: buffer,
        })
```

Response is:

```
{
  statusCode: '406',
  error: 'Not Acceptable',
  message: 'the request is not multipart'
}
```

I tried setting my `Content-Type` to `multipart/form-data`, but that didn't do much.

**Comment by inian** [comment, marked as answer]
Hi @creativiii, the binary upload interface was implemented only for createObjects. I have now implemented it for updateObjects as well [here](https://github.com/supabase/storage-api/pull/34). This will get deployed in a couple of days and your code should work.

**Reply by leo-petrucci** [reply]
That's fantastic, thank you!

**Reply by inian** [reply]
This is deployed now

**Reply by ox-harris** [reply]
This is method is giving me strange results: the upload seems to be successful, but visiting the file's public URL, all I see (of the upload image) is something like a blank 20px by 20px image. I can't even guess whats wrong.
